library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -82,13 +84,41 @@ def self.configured?
       @configured = out["exit"] == 0
     end
 
-    # Creates a new 'single' snapshot
+    # Returns whether creating the given snapshot type is allowed
+    # Information is taken from Linuxrc (DISABLE_SNAPSHOTS)
+    #   * "all" - all snapshot types are temporarily disabled
+    #   * "around" - before and after calling YaST
+    #   * "single" - single snapshot at a given point
+    #
+    # @param [Symbol] one of :around (for :post and :pre snapshots) or :single
+    # @return [Boolean] if snapshot should be created
+    def self.create_snapshot?(snapshot_type)
+      disable_snapshots = Yast::Linuxrc.get_value(Yast::LinuxrcClass::DISABLE_SNAPSHOTS)
```

Comment:
Linuxrc has nothing to do with snapshots, it even doesn't know the feature. Adding just another API doesn't look like a good idea. Code should be close to other similar functionality. it might be worth refactoring, but that is out of scope of this PR, IMO.